Revit_Core_Engine/Create/View/ViewSection/ElevationByLine.cs

```diff
@@ -0,0 +1,80 @@
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using Autodesk.Revit.DB;
+using BH.Engine.Geometry;
+using BH.oM.Adapters.Revit;
+using BH.oM.Base.Attributes;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+
+namespace BH.Revit.Engine.Core
+{
+    public static partial class Create
+    {
+        /***************************************************/
+        /****               Public Methods              ****/
+        /***************************************************/
+
+        [Description("Creates a single Elevation view based on given line. Elevation marker will be placed in the middle of the line.")]
+        [Input("elevationLine", "Location line of the elevation view and the bottom line for the crop region shape.")]
+        [Input("elevationName", "Name of the elevation view.")]
+        [Input("referenceViewPlan", "ViewPlan that elevation marker is referenced to.")]
+        [Input("depth", "Depth of the elevation view.")]
+        [Input("height", "Height of the elevation view.")]
+        [Input("offset", "Offset that will be added to the elevation view CropBox shape.")]
+        [Input("elevationMarker", "Elevation marker that has been created (out parameter).")]
+        [Input("viewTemplateId", "Optional, the View Template Id to be applied in the view.")]
+        [Input("cropRegionVisible", "True if the crop region should be visible.")]
+        [Input("annotationCrop", "True if the annotation crop should be visible.")]
+        [Output("elevationView", "Elevation view that has been created.")]
+        public static ViewSection ElevationByLine(this Line elevationLine, string elevationName, View referenceViewPlan, double depth, double height, double offset, out ElevationMarker elevationMarker, ElementId viewTemplateId = null, bool cropRegionVisible = false, bool annotationCrop = false)
+        {
+            Document doc = referenceViewPlan.Document;
+
+            var elevationMarkerLocation = elevationLine.Evaluate(0.5, true);
+            var lineDirection = elevationLine.Direction.Normalize();
+            var markerDirection = lineDirection.VectorFromRevit().Rotate(-Math.PI / 2, BH.oM.Geometry.Vector.ZAxis).ToRevit();
+            var angle = -markerDirection.AngleTo(XYZ.BasisX);
+
+            if (Math.Abs(angle) < Tolerance.Angle)
```

Comment:
I would check against other rotations as well - see suggestion below (but written off the top of my head, testing needed plus probably can be done smarter). Alternatively you can check based on dot product with global X and Y, the code should be simpler then?
```suggestion
            double absAngle = Math.Abs(angle);
            double refAngle = 0;
            while (refAngle - Tolerance.Angle <= absAngle)
            {
                if (Math.Abs(absAngle - refAngle) < Tolerance.Angle)
                    {
                        if (angle > 0)
                            angle = refAngle;
                        else
                            angle = -refAngle;
                        
                        break;
                    }
                
                refAngle += Math.PI;
            }
```

Reply:
I fixed angle calculation 👍